allennlp/data/token_indexers/pretrained_transformer_indexer.py

```diff
@@ -9,16 +9,17 @@
 from allennlp.data.vocabulary import Vocabulary
 from allennlp.data.tokenizers.token import Token
 from allennlp.data.token_indexers.token_indexer import TokenIndexer
-
+from allennlp.data.token_indexers import SingleIdTokenIndexer
 
 logger = logging.getLogger(__name__)
 
 
 @TokenIndexer.register("pretrained_transformer")
-class PretrainedTransformerIndexer(TokenIndexer[int]):
+class PretrainedTransformerIndexer(SingleIdTokenIndexer):
     """
-    This :class:`TokenIndexer` uses a tokenizer from the ``transformers`` repository to
-    index tokens.  This ``Indexer`` is only really appropriate to use if you've also used a
+    This ``TokenIndexer`` assumes that Tokens already have their indexes in them.
```

Comment:
Is it possible to assert the indices are present in the tokens already? If not, see my comment about a test to distinguish between tokens that already have indices and those that don't.

Reply:
Added a check to the indexer

Reply:
Adding this check also spoiled potential bugs in masked_lm and casual_lm readers.